tests/backends/data/test_es.py

```diff
@@ -0,0 +1,636 @@
+from elasticsearch import ApiError
+from elasticsearch import ConnectionError as ESConnectionError
+from elasticsearch import Elasticsearch
+from elasticsearch.helpers import ScanError, bulk
+
+from ralph.backends.data.base import BaseOperationType, DataBackendStatus
+from ralph.backends.data.es import ESDataBackend, ESDataBackendSettings, ESQuery
+from ralph.exceptions import BackendException, BackendParameterException
+
+from tests.fixtures.backends import (
+    ES_TEST_FORWARDING_INDEX,
+    ES_TEST_INDEX,
+    get_es_fixture,
+)
+
+
+def test_backends_data_es_data_backend_default_instantiation(monkeypatch, fs):
+    """Test the `ESDataBackend` default instantiation."""
+    # pylint: disable=invalid-name
+    fs.create_file(".env")
+    backend_settings_names = [
+        "ALLOW_YELLOW_STATUS",
+        "CLIENT_OPTIONS",
+        "CLIENT_OPTIONS__ca_certs",
+        "CLIENT_OPTIONS__verify_certs",
+        "DEFAULT_CHUNK_SIZE",
+        "DEFAULT_INDEX",
+        "HOSTS",
+        "LOCALE_ENCODING",
+    ]
+    for name in backend_settings_names:
+        monkeypatch.delenv(f"RALPH_BACKENDS__DATA__ES__{name}", raising=False)
+
+    assert ESDataBackend.name == "es"
+    assert ESDataBackend.query_model == ESQuery
+    assert ESDataBackend.default_operation_type == BaseOperationType.INDEX
+    assert ESDataBackend.settings_class == ESDataBackendSettings
+    backend = ESDataBackend()
+    assert backend.default_index == "statements"
+    assert backend.locale_encoding == "utf8"
+    assert isinstance(backend.client, Elasticsearch)
+    elasticsearch_node = backend.client.transport.node_pool.get()
+    assert elasticsearch_node.config.ca_certs is None
+    assert elasticsearch_node.config.verify_certs is None
+    assert elasticsearch_node.host == "localhost"
+    assert elasticsearch_node.port == 9200
+
+
+def test_backends_data_es_data_backend_instantiation_with_settings():
+    """Test the `ESDataBackend` instantiation with settings."""
+    settings = ESDataBackendSettings(
+        ALLOW_YELLOW_STATUS=False,
+        CLIENT_OPTIONS={"verify_certs": True, "ca_certs": "/path/to/ca/bundle"},
+        DEFAULT_CHUNK_SIZE=500,
+        DEFAULT_INDEX=ES_TEST_INDEX,
+        HOSTS=["https://elasticsearch_hostname:9200"],
+        LOCALE_ENCODING="utf-16",
+    )
+    backend = ESDataBackend(settings)
+    assert backend.default_index == ES_TEST_INDEX
+    assert backend.locale_encoding == "utf-16"
+    assert isinstance(backend.client, Elasticsearch)
+    elasticsearch_node = backend.client.transport.node_pool.get()
+    assert elasticsearch_node.config.ca_certs == Path("/path/to/ca/bundle")
+    assert elasticsearch_node.config.verify_certs is True
+    assert elasticsearch_node.host == "elasticsearch_hostname"
+    assert elasticsearch_node.port == 9200
+
+    try:
+        ESDataBackend(settings)
+    except Exception as err:  # pylint:disable=broad-except
+        pytest.fail(f"ESDataBackend should not raise exceptions: {err}")
```

Comment:
We should be consistent between backend tests. I don't get why we care about checking if an exception occurs in this test. Shouldn't we use the `es_backend` fixture ? cf [mongoDB test](https://github.com/openfun/ralph/pull/372/files#diff-580781a4f0491f1f6b6f56d67c9ca90002a16eaa4e0ecc36cb09573492bf3963R56)

Reply:
Well spotted! Thanks) I have forgotten this test in the MongoDB pull request, I will update it.
The idea of the test is to check that instantiating a second instance of the backend does not raise an exception.
Do we update the error message to something like: "Two instances of ESDataBackend should not raise exceptions"?